lib/galaxy/datatypes/sequence.py

```diff
@@ -940,4 +946,84 @@ def sniff(self, filename):
                             pairs = True
                     if seq and coor and pairs:
                         return True
+        return False
+
+
+class DotBracket ( Sequence ):
+    edam_format = "format_1457"
+    file_ext = "dbn"
+    
+    sequence_regexp = re.compile( "^[ACGTURYKMSWBDHVN]+$", re.I)
+    structure_regexp = re.compile( "^[\(\)\.\[\]]*" )
+    
+    def set_meta( self, dataset, **kwd ):
+        """
+        Set the number of sequences and the number of data lines
+        in dataset.
+        """
+        if self.max_optional_metadata_filesize >= 0 and dataset.get_size() > self.max_optional_metadata_filesize:
+            dataset.metadata.data_lines = None
+            dataset.metadata.sequences = None
+            dataset.metadata.seconday_structures = None
+            return
+        
+        data_lines = 0
+        sequences = 0
+        
+        for line in file( dataset.file_name ):
+            line = line.strip()
+            data_lines += 1
+            
+            if line and line.startswith( '>' ):
+                sequences += 1
+        
+        dataset.metadata.data_lines = data_lines
+        dataset.metadata.sequences = sequences
+    
+    def sniff(self, filename):
+        """
+        The Dot-Bracket format is as follows:
+        
+        >sequenceName1
+        CCCaaaGGG
+        (((...)))
+        >sequenceName2
+        GGGuuuCCC
+        (((...)))
+        
+        Because it remains unclear whether the Dot-Bracket format may
+        contain multiple sequences per file, sniffing is only applied
+        on the first 3 lines.
+        """
+        
+        i = 0
+        
+        with open( filename, "r" ) as handle:
+            for line in handle:
+                line = line.strip()
+                
+                state = i % 3
```

Comment:
Thanks! This is also what we are facing more or less. The format has no many derivatives. Especially, the last one is really annoying :)